src/http/Message.h

```diff
@@ -143,9 +144,11 @@ class Message : public RefCountable
 
 } // namespace Http
 
-inline void
-HTTPMSGUNLOCK(Http::Message *a)
+template <class T>
+void
+HTTPMSGUNLOCK(T *&a)
 {
+    static_assert(std::is_base_of<Http::Message, T>::value, "T must inherit from Http::Message");
```

Comment:
Nice! What kind of compiler error message does a developer get if she accidentally feeds an HttpReply::Pointer object to the new HTTPMSGUNLOCK()?

Reply:
```
error: no matching function for call to ‘HTTPMSGUNLOCK(const HttpReplyPointer&)’
  HTTPMSGUNLOCK(reply_);
                      ^
MemObject.cc:195:22: note: candidate is:
In file included from HttpRequest.h:17:0,
                 from HttpReply.h:14,
                 from MemObject.cc:15:
http/Message.h:149:1: note: template<class M> void HTTPMSGUNLOCK(M*&)
 HTTPMSGUNLOCK(M *&a)
 ^
http/Message.h:149:1: note:   template argument deduction/substitution failed:
MemObject.cc:195:22: note:   mismatched types ‘M*’ and ‘const HttpReplyPointer {aka const RefCount<HttpReply>}’
  HTTPMSGUNLOCK(reply_);
```

Reply:
What kind of compiler error message does a developer get if she accidentally feeds an `int *` or a similar static_assert-reaching argument to the new HTTPMSGUNLOCK()?

Reply:
1. int*:
```
MemObject.cc:236:25: error: request for member ‘unlock’ in ‘* a’, which is of non-class type ‘int’
         if (a->unlock() == 0)
```

2. A class with unlock() which is not a child of Http::Message:
```
MemObject.cc:240:5: error: static assertion failed: M must inherit from Http::Message
     static_assert(std::is_base_of<Http::Message, M>::value, "M must inherit from Http::Message");
```

Reply:
Thank you. I think this is good enough.